src/components/ComputeModal.js

```diff
@@ -844,6 +867,24 @@ export const ComputeModalBase = ({ onDismiss, onSuccess, runtimes, persistentDis
                 })
               ])
             ])
+          ]),
+          h(IdContainer, [
+            id => div({ style: { gridColumnEnd: 'span 3', marginTop: '0.5rem' } }, [
+              label({ htmlFor: id, style: styles.label }, ['Location']),
+              h(InfoBox, { style: { marginLeft: '0.5rem' } }, [
+                'By default, workflow and cloud environment VMs will run in the same region as the workspace bucket. ',
+                'Take care as overriding the default can lead to network egress charges.'
+              ]),
+              div({ style: { marginTop: '0.5rem' } }, [
+                h(Select, {
+                  id,
+                  isSearchable: false,
+                  value: computeConfig.computeRegion,
+                  onChange: ({ value, locationType }) => onLocationChange(value, locationType),
+                  options: _.flow(_.filter(l => l.value !== 'US'), _.sortBy('label'))(allRegions)
+                })
+              ])
+            ])
           ])
```

Comment:
We talked about it offline but just wanted to call out here too that where this component is, will be occupied by a `Link` for Spark UI once the [corresponding PR](https://github.com/DataBiosphere/terra-ui/pull/2643) is in and we were hoping this component can be slated below `Compute type`?